logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt

```diff
@@ -0,0 +1,26 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ClientId
+import com.wire.kalium.logic.functional.suspending
+
+interface CurrentClientIdUseCase {
+    suspend operator fun invoke(): CurrentClientIdResult
+}
+
+class CurrentClientIdUseCaseImpl(private val clientRepository: ClientRepository) : CurrentClientIdUseCase {
```

Comment:
Do you actually need the clientID? My first though is that the consumer shouldn't deal with that and we can instead have a `NeedsToRegisterClientUseCase` which returns a boolean.

Reply:
That's a good point, but we already have `SelfClientsUseCase` so the consumer can already get even more than just `ClientId` 🤔 @typfel let me know if I should still change it to `NeedsToRegisterClientUseCase` 😉

Reply:
Well depends, do you intend to use the client ID where you use the use case or will you just throw it away and look for the error? If it's the latter I think it cleaner and more readable to have a `NeedsToRegisterClientUseCase`.

Reply:
You're right, I changed it to `NeedsToRegisterClientUseCase` which returns a boolean.